src/lib/util/Limiter.ts

```diff
@@ -0,0 +1,61 @@
+import { createMethodDecorator } from 'klasa-decorators';
+import { Collection, Constructor } from 'discord.js';
+import { RateLimitManager } from 'klasa';
+import { createClassDecorator } from './util';
+import { Time } from './constants';
+
+export const enum LimitErrors {
+	Ratelimited = 'RATELIMIT'
+}
+
+export interface LimitedClass {
+	ratelimits: Collection<string, RateLimitManager>;
+	limiter: boolean;
+}
+
+export interface MethodRatelimitedError {
+	error: LimitErrors.Ratelimited;
+	remainingTime: number;
+}
+
+export function classLimitInitialization() {
+	return createClassDecorator((target: Constructor<LimitedClass>) => class extends target {
+
+		public ratelimits: Collection<string, RateLimitManager> = new Collection();
```

Comment:
If the `implements` helps TypeScript inferring the type of the arguments, remove their type casts:
```suggestion
		public ratelimits = new Collection();
```

But if it doesn't:
```suggestion
		public ratelimits = new Collection<string, RateLimitManager>();
```